src/hardware/adlib.cpp

```diff
@@ -50,19 +50,18 @@ namespace OPL2 {
 			return val;
 		}
 
-		virtual void Generate( MixerChannel* chan, Bitu samples ) {
-			Bit16s buf[1024];
+	        virtual void Generate(MixerChannel *chan, uint16_t samples)
+	        {
+		        Bit16s buf[1024];
 			while( samples > 0 ) {
-				Bitu todo = samples > 1024 ? 1024 : samples;
-				samples -= todo;
+			        const auto todo = samples > 1024 ? 1024 : samples;
```

Comment:
Note for reviewers: `samples` never exceeds 32KB, because the mixer interface at-most passes a 16-bit request size into the audio device callback.